### ISSUE TITLE
Stabilize shared-vs-differentiated quadratic architecture algebra

### ISSUE THREAD

**Issue by zuizui0223** [zuizui0223/bita#203]
## Reproduced core algebra dependence on trait/fitness units

`trait_architecture/differentiation.py` is the original shared-axis versus differentiated-axis quadratic model, but it still forms raw products such as `w1*w2`, `weight*optimum`, `conflict**2`, and a three-term differentiated optimum numerator. Its default `neutral_tolerance=1e-12` is also silently treated as a physical fitness-unit band when callers do not explicitly request one.

The same quadratic model has a simpler stable representation. Let `m` be the shared weighted optimum, `L` the shared conflict load, and `s` the decoupling fraction. Then

```text
x* - y* = s (theta1-theta2)
residual differentiated loss = (1-s)L
recoverable loss = sL
architecture gain = sL-K
```

Together with the weighted-center constraint, these identities reconstruct both differentiated optima without the raw dimensional products.

## Repair contract

- compute weighted means from normalized weight fractions;
- compute conflict load with a stable harmonic coefficient and multiplication order that avoids explicit `distance**2`;
- compute decoupling and its complement from the harmonic conflict coefficient versus coupling;
- reconstruct differentiated optima from the weighted center and retained separation;
- compute residual/recoverable losses and architecture gain from `L` and `s` rather than subtracting large optimized fitness values;
- make the *default* preference classification use a roundoff-only band relative to `(threshold,cost)`;
- preserve an explicitly supplied `neutral_tolerance` as the existing absolute fitness-unit neutral band;
- add regressions under common fitness scaling and consistent trait-coordinate/weight scaling, including a strict tiny gain below the old `1e-12` default floor.

No quadratic model, differentiation threshold, or explicit user-declared neutral-band semantics change.
